src/transformStateWithClones.js

```diff
@@ -7,7 +7,29 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const currentState = { ...state };
+
+  for (let i = 0; i < actions.length; i++) {
+    switch (actions[i].type) {
+      case 'addProperties':
+        result[i] = { ...Object.assign(currentState, actions[i].extraData) };
+        break;
+      case 'removeProperties':
+        for (let y = 0; y < actions[i].keysToRemove.length; y++) {
+          delete currentState[actions[i].keysToRemove[y]];
+        }
+        result[i] = { ...currentState };
+        break;
+
+      case 'clear':
+        Object.keys(currentState).forEach((key) => delete currentState[key]);
+        result[i] = { ...currentState };
+        break;
+    }
```

Comment:
[CODE STYLE]: Use a switch statement if you have a limited amount of conditions, which you have done correctly. However, ensure that you're not performing unnecessary operations within each case that could be done once outside of the switch.